bool initialized_
int32 data